### PR TITLE
Add support for GeoSpatial points in string format

### PR DESCRIPTION
**Pull request by abhinavdangeti**
Following specification used in elastic search:
    https://www.elastic.co/guide/en/elasticsearch/reference/current/geo-point.html

Also see:
    https://issues.couchbase.com/browse/MB-30542

### CODE REVIEW

File: geo/parse.go [hunk 1]
```diff
@@ -16,14 +16,19 @@ package geo
...
+
+	"github.com/mmcloughlin/geohash"
 )
 
 // ExtractGeoPoint takes an arbitrary interface{} and tries it's best to
 // interpret it is as geo point.  Supported formats:
 // Container:
 // slice length 2 (GeoJSON)
 //  first element lon, second element lat
+// string (coordinates seperated by comma, or a geohash)
```
Comment (sreekanth-cb) [inline geo/parse.go]: typo on seperated?

File: geo/parse.go [hunk 2]
```diff
@@ -59,6 +64,35 @@ func ExtractGeoPoint(thing interface{}) (lon, lat float64, success bool) {
...
+					foundLat = true
+				}
+				lon, err = strconv.ParseFloat(points[1], 64)
+				if err == nil {
+					foundLon = true
+				}
+			}
+		} else {
+			// geo hash
+			lat, lon = geohash.Decode(geoStr)
```
Comment (sreekanth-cb) [inline geo/parse.go]: worth having a few UTs to test the negative cases for this Decode() fn?,
For eg: like pass couple of invalid hashes,  
Another thing to see is that how that lat/lon values returned on an invalid hash percolates to later sections of the logic.

File: geo/parse.go [hunk 3]
```diff
@@ -16,14 +16,19 @@ package geo
...
+	"github.com/mmcloughlin/geohash"
 )
 
 // ExtractGeoPoint takes an arbitrary interface{} and tries it's best to
 // interpret it is as geo point.  Supported formats:
 // Container:
 // slice length 2 (GeoJSON)
 //  first element lon, second element lat
+// string (coordinates seperated by comma, or a geohash)
+//  first element lat, second element lon
```
Comment (sreekanth-cb) [inline geo/parse.go]: minor extra space? :)

Comment (abhinavdangeti) [reply geo/parse.go]: That extra space before first, is following the format from before.

Per elasticsearch spec here: https://www.elastic.co/guide/en/elasticsearch/reference/current/geo-point.html

With slices, its lon first and lat after, but with strings its lat first and lon after - that's what the comment(s)^ above talk about.

File: geo/parse.go (hunk 2 above)
Comment (abhinavdangeti) [reply geo/parse.go]: Well..  geohash is a library, that already has it's own suite of unit tests.

Comment (sreekanth-cb) [reply geo/parse.go]: geohash is a library -> its always a debatable thing about the robustness or maturity of the 3rd party libs.  
Main wrinkle points for me here are,
1. How confident are you that, the values returned from that lib for an invalid hash are valid Or that it won't break any of the remaining code in bleve? 

2. Having few unit tests to set/validate the response expectations from a library helps in confirming minimum sanity at least during a SHA bump for it at a later point in time for accommodating a genuine feature or bug fix on their side.

**Comment by abhinavdangeti** [comment]
@steveyen @sreekanth-cb Ping for re-review. The latest patch borrows only the necessary functionality from https://github.com/mmcloughlin/geohash for decoding geohashes.

**Comment by abhinavdangeti** [comment]
Ping for re-review @sreekanth-cb .

**Pull request merged**